torch_xla/csrc/translator.cpp

```diff
@@ -162,6 +163,478 @@ class ComputationContext {
+  xla::XlaOp xla_output = BuildSum(node, cctx->OpForInput(node, 0));
+  cctx->AddNodeOp(node, xla_output);
+}
+
+void TranslateNllLoss(const torch::jit::Node* node, ComputationContext* cctx,
+                      xla::PrecisionConfig::Precision /*conv_precision*/,
+                      xla::XlaBuilder* /*b*/) {
+  XLA_CHECK_EQ(node->inputs().size(), 5);
+  xla::XlaOp xla_output =
+      BuildNllLoss(node, cctx->OpForInput(node, 0), cctx->OpForInput(node, 1));
+  cctx->AddNodeOp(node, xla_output);
+}
+
+void TranslateNllLossBackward(
+    const torch::jit::Node* node, ComputationContext* cctx,
+    xla::PrecisionConfig::Precision /*conv_precision*/,
+    xla::XlaBuilder* /*b*/) {
+  XLA_CHECK_EQ(node->inputs().size(), 7);
+  xla::XlaOp xla_output = BuildNllLossBackward(node, cctx->OpForInput(node, 1),
+                                               cctx->OpForInput(node, 2));
+  cctx->AddNodeOp(node, xla_output);
+}
+
+void TranslateSize(const torch::jit::Node* node, ComputationContext* cctx,
+                   xla::PrecisionConfig::Precision /*conv_precision*/,
+                   xla::XlaBuilder* /*b*/) {
+  XLA_CHECK_EQ(node->inputs().size(), 1);
+  std::vector<xla::int64> size_op_result;
+  xla::XlaOp xla_output =
+      BuildSize(node, cctx->OpForInput(node, 0), &size_op_result);
+  cctx->AddSizeOpResult(node->output(0), size_op_result);
+  cctx->AddNodeOp(node, xla_output);
+}
+
+void TranslateConstant(const torch::jit::Node* node, ComputationContext* cctx,
+                       xla::PrecisionConfig::Precision /*conv_precision*/,
+                       xla::XlaBuilder* b) {
+  cctx->AddNodeOp(node, GetConstantOp(b, node));
+}
+
+void TranslateUndefined(const torch::jit::Node* node, ComputationContext* cctx,
+                        xla::PrecisionConfig::Precision /*conv_precision*/,
+                        xla::XlaBuilder* /*b*/) {
+  cctx->AddUndefinedInput(ComputationContext::OutputId(node));
+}
+
+void TranslateSumToSize(const torch::jit::Node* node, ComputationContext* cctx,
+                        xla::PrecisionConfig::Precision /*conv_precision*/,
+                        xla::XlaBuilder* /*b*/) {
+  XLA_CHECK_EQ(node->inputs().size(), 2);
+  xla::XlaOp xla_output =
+      BuildSumToSize(node, cctx->OpForInput(node, 0), cctx->GetSizeOpValues());
+  cctx->AddNodeOp(node, xla_output);
+}
+
+void TranslateNop(const torch::jit::Node* /*node*/,
+                  ComputationContext* /*cctx*/,
+                  xla::PrecisionConfig::Precision /*conv_precision*/,
+                  xla::XlaBuilder* /*b*/) {}
+
+using TranslationHandler = std::function<void(
+    const torch::jit::Node* node, ComputationContext* cctx,
+    xla::PrecisionConfig::Precision conv_precision, xla::XlaBuilder* b)>;
+
+std::unordered_map<torch::jit::NodeKind, TranslationHandler>*
+TranslationHandlers() {
+  static auto t =
+      new std::unordered_map<torch::jit::NodeKind, TranslationHandler>();
+  (*t)[at::aten::add] = TranslateArithmetic;
```

Comment:
You are going to re-init this every time you call this function ;)

A* Create() { ... }

A* Get() {
  static A* a = Create();
  return a;
}

Reply:
`t` is static, what am I missing?

Reply:
Nevermind, it's obvious. Let me fix it.